File: tests/python/cases/case_071_snapshot_profile.py
"""
Task case 071: snapshot profile payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_071_snapshot_profile(input_data):
    # Task: snapshot profile records and prepare transport-ready payload.
    task_name = 'snapshot_profile'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'profile',
        'stage': 'snapshot',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_071'] = payload
    return 'ok'

File: tests/nodejs/cases/case_071_snapshot_profile.js
/**
 * Task case 071: snapshot_profile
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case071SnapshotProfile(input) {
  // Task: snapshot profile records and prepare transport-ready payload.
  const taskName = 'snapshot_profile';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'profile',
    stage: 'snapshot',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_071', payload);
  return 'ok';
}

module.exports = {
  case071SnapshotProfile,
};
